tests/integration/data_sources/test_local_files.py

```diff
@@ -32,3 +44,108 @@ def test_sample_dataset(self, local_files_dataset: Dataset) -> None:
+
+        # Primary purpose of this test is to make sure LocalFiles can import the data,
+        # but here we also verify that it has been encoded based on the desired
+        # coordinate mode.
+        with (src_data_dir / "data.geojson").open() as f:
+            fc = json.load(f)
+            feat_x, feat_y = fc["features"][0]["geometry"]["coordinates"]
+        if coordinate_mode == GeojsonCoordinateMode.CRS:
+            assert feat_x == pytest.approx(
+                features[0].geometry.shp.x * custom_projection.x_resolution
+            )
+            assert feat_y == pytest.approx(
+                features[0].geometry.shp.y * custom_projection.y_resolution
+            )
+        elif coordinate_mode == GeojsonCoordinateMode.WGS84:
+            assert feat_x == pytest.approx(seattle.shp.x)
+            assert feat_y == pytest.approx(seattle.shp.y)
+
+        dataset_config = {
+            "layers": {
+                "local_file": {
+                    "type": "vector",
+                    "data_source": {
+                        "name": "rslearn.data_sources.local_files.LocalFiles",
+                        "src_dir": src_data_dir.path,
+                    },
+                },
+            },
+            "tile_store": {
+                "name": "file",
+                "root_dir": "tiles",
+            },
+        }
+        with open(os.path.join(tmp_path, "config.json"), "w") as f:
+            json.dump(dataset_config, f)
+
+        # Make the window in a different projection.
+        # First window has same coordinates but it's in a different projection so if
+        # GeoJSON is handled correctly, it shouldn't match with any items.
+        # Second window has the correctly aligned coordinates.
+        window_projection = Projection(CRS.from_epsg(32610), 5, -5)
+
+        window_center = seattle.to_projection(custom_projection).shp
+        bad_window = Window(
+            path=Window.get_window_root(ds_path, "default", "bad"),
+            group="default",
+            name="bad",
+            projection=window_projection,
+            bounds=(
+                int(window_center.x) - 10,
+                int(window_center.y) - 10,
+                int(window_center.x) + 10,
+                int(window_center.y) + 10,
+            ),
+            time_range=None,
+        )
+        bad_window.save()
+
+        window_center = seattle.to_projection(window_projection).shp
+        good_window = Window(
+            path=Window.get_window_root(ds_path, "default", "good"),
+            group="default",
+            name="good",
+            projection=window_projection,
+            bounds=(
+                int(window_center.x) - 10,
+                int(window_center.y) - 10,
+                int(window_center.x) + 10,
+                int(window_center.y) + 10,
+            ),
+            time_range=None,
+        )
+        good_window.save()
```

Comment:
Suggestion: Move the good and bad window to pytest fixtures if possible as well as the dataset config